pkg/rabbit/setup.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rabbit
+
+import (
+	"time"
+
+	"github.com/NeowayLabs/wabbit"
+	wabbitamqp "github.com/NeowayLabs/wabbit/amqp"
+	"go.uber.org/zap"
+)
+
+var (
+	retryCounter                                    = 0
+	cycleDuration                                   = 2
+	cleaningUp                                      = false
```

Comment:
If the plan is to reuse this package, we can't have a global variables like this

Reply:
hmmm they are private but what do you suggest?

Reply:
It's more like if you have A and B using this service simultaneously, they'll both be clobbering each others states. Was suggesting to move them into a struct but looks like you just did that 👍